distgo/paramproduct.go

```diff
@@ -23,6 +23,9 @@ type ProductParam struct {
 	// the configuration.
 	ID ProductID
 
+	// Name is the name of this product. If not specified, the ID is generally used.
+	Name *string
```

Comment:
Prefer making this non-optional and explicit:

```suggestion
	// Name is the name of this product. If it is not specified in configuration, defaults to the
	// same value as ID. The Name field should be used for operations that require the product's
	// name for configuration or rendering. The field is intended to be used in cases when the
	// product name used for these purposes cannot be the ID -- for example, if there are different
	// products that want to render output with the same logical name.
	Name string
```